client/mgcp/driver/src/main/java/org/restcomm/media/client/mgcp/stack/JainMgcpStackImpl.java

```diff
@@ -456,6 +463,7 @@ public void activate() {
          */
         private void shutdown() {
             this.active = false;
+            this.interrupt();
```

Comment:
I think it will be enough to set active=false and let the thread elegantly terminate by itself.